alf/experience_replayers/replay_buffer.py

```diff
@@ -196,17 +359,20 @@ def __init__(self,
                  num_environments,
                  max_length=1024,
                  device="cpu",
+                 allow_multiprocess=False,
                  name="ReplayBuffer"):
         super().__init__(
             data_spec,
             num_environments,
             max_length=max_length,
             device=device,
+            allow_multiprocess=allow_multiprocess,
             name=name)
 
-    def add_batch(self, batch, env_ids=None):
-        self.enqueue(batch, env_ids)
+    def add_batch(self, batch, env_ids=None, blocking=False):
+        self.enqueue(batch, env_ids, blocking=blocking)
 
+    @atomic
```

Comment:
Is there readwrite lock? If so, we can just use readlock for reading operations so that multiple reading operations do not block each other.

Reply:
According to https://stackoverflow.com/questions/53119297/python-multiprocessing-read-write-lock
We need to write our own lock.
Do you think it's good to have for us?
For replayer, writing is once per time step, so will be the most frequent operation, while reading will be slower and less frequent.  For queue, probably reading and writing will be equally frequent, both being fairly small amounts of data.

Addressed the other comments.